test/e2e/network/network_policy.go

```diff
@@ -1890,69 +1890,66 @@ var _ = SIGDescribe("NetworkPolicy [Feature:SCTPConnectivity][LinuxOnly][Disrupt
 })
 
 func testCanConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int) {
-	testCanConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP)
+	testConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP, 3, false)
 }
 
 func testCannotConnect(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int) {
-	testCannotConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP)
+	testConnectProtocol(f, ns, podName, service, targetPort, v1.ProtocolTCP, 3, true)
 }
 
 func testCanConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol) {
-	ginkgo.By(fmt.Sprintf("Creating client pod %s that should successfully connect to %s.", podName, service.Name))
-	podClient := createNetworkClientPod(f, ns, podName, service, targetPort, protocol)
-	defer func() {
-		ginkgo.By(fmt.Sprintf("Cleaning up the pod %s", podClient.Name))
-		if err := f.ClientSet.CoreV1().Pods(ns.Name).Delete(context.TODO(), podClient.Name, metav1.DeleteOptions{}); err != nil {
-			framework.Failf("unable to cleanup pod %v: %v", podClient.Name, err)
-		}
-	}()
-	checkConnectivity(f, ns, podClient, service)
+	testConnectProtocol(f, ns, podName, service, targetPort, protocol, 3, false)
 }
 
 func testCannotConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol) {
-	ginkgo.By(fmt.Sprintf("Creating client pod %s that should not be able to connect to %s.", podName, service.Name))
-	podClient := createNetworkClientPod(f, ns, podName, service, targetPort, protocol)
+	testConnectProtocol(f, ns, podName, service, targetPort, protocol, 3, true)
+}
+func testConnectProtocol(f *framework.Framework, ns *v1.Namespace, podName string, service *v1.Service, targetPort int, protocol v1.Protocol, tryCount int, expectErr bool) {
```

Comment:
is the `tryCount` truly something we need to expose? Or can we just internally retry `n` times for `n = 3` here. I don't see any of the tests passing anything other than a `tryCount` of `3`. Especially, if this is a short-term tactical thing till we get the status field added in the policy object.

Reply:
Perhaps not. In my first iteration, I had the `tryCount` set to 1 if there were no network policies created, which is the case in `BeforeEach` , where the server is created and clients are created to verify connectivity on ports 80 and 81. I'll go ahead and just remove that parm, will push update shortly. And, as you say, this is really a short-term fix until we get the status field.

Reply:
Done.